ncm-dirperm/src/main/perl/dirperm.pm

```diff
@@ -185,14 +206,35 @@ sub process_path {
 			open TMP, ">>$path";
 			close TMP;
 			if (! -f $path) {
-				$self->error("error creating file $path");
+				$self->error("Error creating file $path");
 				return 0;
 			}
 		} elsif ($type eq "d") {
+            # Should we check if the needed mount is actually mounted?
+            # (i.e. don't build path on / (root) if the mount point is not available)
+            if ($pathentry->{checkmount}) {
+                # Which mountpoint is path built on?
+                my @path_mountpoints = grep {$path =~ m/^$_\//} @configured_mounts;
+
+                if (!@path_mountpoints) {
+                    $self->error("Couldn't find matching mount in profile for $path. Skipping.");
+                    return 0;
+                } elsif (not exists($available_mounts{$path_mountpoints[0]})) {
```

Comment:
why only check the first element? shouldn't it be the "longest" path? suppoes there is 3 partitions mounted under
`/`, `/var` and `/var/tmp`, then if you are using dirperm on `/var/tmp/x`, the `/var/tmp mountpoint` is the one to check, no? (or all of them?)